app/assets/javascripts/admin/controllers/tree_app.coffee

```diff
@@ -1,51 +1,34 @@
-#admin
-#admin
-#admin
-#admin
-#admin
-#admin
-#admin
-#admin
-
-
-
 app = angular.module('treeApp')
 
 app.controller 'BodyController', ($scope, Order, OrderItem) ->
   $scope.cart = Order.cart()
 
   #  This add_product uses item.id for the product page.
   $scope.add_product = (item) ->
+    if item.quantity == 0
+      $scope.remove = (item) ->
```

Comment:
I don't think this would work as expected. You're just creating a function named ```$scope.remove(item)``` not actually deleting anything